Refactor proxy reset (#1253)
kyma-project/istio: docs/contributor/04-10-technical-design.md

The SidecarsRestarter is responsible for keeping the proxy sidecars in the desired state. It restarts Pods that are in the `Running` state, are part of the service mesh, and have the annotation `sidecar.istio.io/status`.
The Istio CR and the [Istio version](#istio-version) represent the desired state. Pods are restarted in chunks with limits on the number that can be restarted in one reconciliation and the number that can be listed when requesting from the Kubernetes API Server. If the number of Pods that must be restarted exceeds the limits, it happens in the next reconciliation. In such a case, the reconciliation request is requeued with a 1-minute delay to allow time for the Kubernetes scheduler to restart the Deployments.

> Restarting sidecars is devided in two phases:
> - In the first phase only Kyma workloads are restarted. Determining Kyma workloads is done by checking if workload is running in `kyma-system` namespace or is having the `kyma-project.io/module` annotation. All Kyma workloads are restarted without pagination and if there is a problem with the restart Istio CR will result in `Error` state and will requeue reconciliation.
> - In the second phase only Customer workloads are restarted. Determining Customer workloads is done by checking if workload is not running in `kyma-system` namespace and is not having the `kyma-project.io/module` annotation. All Customer workloads are restarted with pagination and if there is a problem with the restart Istio CR will result in `Warning` state and will requeue reconciliations with a 1-minute delay.

Reviewer: ```suggestion
Restarting sidecars is divided into two phases:
- In the first phase, only Kyma workloads are restarted. A workload is considered a Kyma workload if it runs in the `kyma-system` namespace or has the `kyma-project.io/module` annotation. All Kyma workloads are restarted without pagination. If there is a problem with the restart, Istio CR is set to the `Error`, and the reconciliation is requeued.
- In the second phase, only customer workloads are restarted. A workload is considered a customer workload if it does not run in the `kyma-system` namespace and does not have the `kyma-project.io/module` annotation. All customer workloads are restarted with pagination. If there is a problem with the restart, Istio CR  is set to the `Warning` state, and the reconciliation is requeued with a 1-minute delay.
```

> During the Customer sidecars restarting phase, the Istio CR remains in the `Processing` state having the following status conditions:

Reviewer: ```suggestion
During the customer sidecars restarting phase, the Istio CR remains in the `Processing` state, having the following status conditions:
```

- The `Ready` condition is set to `false` with the reason `ReconcileRequeued`.
- The `ProxySidecarRestartSucceeded` condition is set to `false` with the reason `ProxySidecarPartiallySucceeded`.

> After completing Customer sidecars restart Istio CR's `Ready` condition is set to `true` and `ProxySidecarRestartSucceeded` is set to `true` with the reason `ProxySidecarRestartSucceeded`.

Reviewer: ```suggestion
After completing the customer sidecars restart, Istio CR's `Ready` condition is set to `true`, and `ProxySidecarRestartSucceeded` is set to `true` with the reason `ProxySidecarRestartSucceeded`.
```
